alerting/src/main/kotlin/org/opensearch/alerting/util/QueryIndexManagement.kt

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.util
+
+import kotlinx.coroutines.CoroutineName
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Mutex
+import org.apache.logging.log4j.LogManager
+import org.opensearch.action.admin.indices.delete.DeleteIndexRequest
+import org.opensearch.action.admin.indices.exists.indices.IndicesExistsRequest
+import org.opensearch.action.admin.indices.exists.indices.IndicesExistsResponse
+import org.opensearch.action.admin.indices.stats.IndicesStatsRequest
+import org.opensearch.action.admin.indices.stats.IndicesStatsResponse
+import org.opensearch.action.support.master.AcknowledgedResponse
+import org.opensearch.alerting.MonitorMetadataService
+import org.opensearch.alerting.model.MonitorMetadata
+import org.opensearch.alerting.opensearchapi.suspendUntil
+import org.opensearch.alerting.settings.AlertingSettings
+import org.opensearch.alerting.util.IndexUtils.Companion.isIndexWriteIndex
+import org.opensearch.client.Client
+import org.opensearch.cluster.ClusterChangedEvent
+import org.opensearch.cluster.ClusterStateListener
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver
+import org.opensearch.cluster.service.ClusterService
+import org.opensearch.common.settings.Settings
+import org.opensearch.index.IndexNotFoundException
+import org.opensearch.threadpool.Scheduler
+import org.opensearch.threadpool.ThreadPool
+
+private val log = LogManager.getLogger(QueryIndexManagement::class.java)
+
+class QueryIndexManagement private constructor(
```

Comment:
There needs to be comments on what kinda of cleanup does this do with the job. What kind of indices qualify for cleanup. Deleted indices and all the other cases.

Reply:
done